CONTRIBUTING.md: Add compilation and instalations steps

beekaz opened the pull request:
Guide on  how to build and install Nmstate from scratch

beekaz committed:
CONTRIBUTING.md: Add compilation and instalations steps

Signed-off-by: beekaz <sooperfect00@gmail.com>

ffmancera commented on CONTRIBUTING.md:
```diff
@@ -67,6 +67,28 @@ settings Git can find.
 
+## Compilation and Installation:
+
+* Fork and git clone (https://github.com/nmstate/nmstate.git)
+`cd nmstate`
+
+* For Fedora 29+
+`sudo dnf install nmstate`
+
+* RHEL 8 using copr repo
+`sudo dnf copr enable nmstate/nmstate-stable`
+`sudo dnf install nmstate`
+
+* For Stables releases
+`python setup.py build`
+`python setup.py install`
+
+* For installation of nmstatectl
+`cargo install nmstatectl`
+
+* To see networking state in yaml format
+`nmstatectl show`
```
The compilation and installations steps we expect for a `CONTRIBUTING.md` file are the ones that a developer would follow. These are the installation steps of an user, not a developer.

In essence, how do you compile the sources of nmstate and install them?

beekaz replied:
Oh my bad, I would and fix that right

liangwen12year replied:
Before running the `make` cmd, make sure that the  system has the Rust compiler installed via https://rustup.rs/.

beekaz replied:
Thank you so much @liangwen12year  I have done that and my nmstatectl  is upto date

cathay4t committed:
Fix cargo clippy

Signed-off-by: Gris Ge <fge@redhat.com>
Signed-off-by: beekaz <sooperfect00@gmail.com>

liangwen12year commented:
Please sign off the commit, the DCO is failing.

cathay4t committed:
New release 2.2.26

=== Breaking changes
 - Deprecate PrettyState class in Python API. (6b410dcf)

=== New features
 - Allowing input multiple desire states for statistics. (9b808821)

=== Bug fixes
 - packaging: Fix rpm build on Fedora. (c40f3ff0)
 - route: warn if the route is missing due to NM delay. (71dedc65)
 - Raise invalid argument error when desired OVS bridge with MAC. (a71eed31)
 - Auto managed ignored OVS port. (c7fd32e8)
 - Fix resolving token in array items for nmpolicy. (58a2fd6d)

Signed-off-by: Gris Ge <fge@redhat.com>
Signed-off-by: beekaz <sooperfect00@gmail.com>

cathay4t committed:
Bump version to 2.2.27

Signed-off-by: Gris Ge <fge@redhat.com>
Signed-off-by: beekaz <sooperfect00@gmail.com>

beekaz committed:
CONTRIBUTING.md: Add compilation and instalations steps

Signed-off-by: beekaz <sooperfect00@gmail.com>

beekaz commented:
> Please sign off the commit, the DCO is failing.

Thank you I have successfully signed off

ihuguet commented on CONTRIBUTING.md:
```diff
@@ -43,6 +43,19 @@ querying network status and applying network states.
+## Installation and Compilation:
+
+- [Fork](https://github.com/nmstate/nmstate.git) and git clone to local
+-[Install rust and Cargo for the development of nmstate](https://doc.rust-lang.org/cargo/getting-started/installation.html) or `apt-get install cargo`
```
Please add a space after the '-'. Also, apt-get is for Ubuntu/Debian only, please put it in a more generic way, like "or install from your distribution's repositories".

ihuguet commented on CONTRIBUTING.md:
```diff
@@ -43,6 +43,19 @@ querying network status and applying network states.
+- [Fork](https://github.com/nmstate/nmstate.git) and git clone to local
+-[Install rust and Cargo for the development of nmstate](https://doc.rust-lang.org/cargo/getting-started/installation.html) or `apt-get install cargo`
+- Change your current directory to the nmstate directory `cd nmstate`
+- Install nmstate `cargo install nmstatectl`
+This will download the necessary dependencies and build the nmstate library and command-line tool
```
This is not correct, I think. This will install the version of nmstatectl that is published in crates.io.
The right way to download the dependencies and build nmstate is `cargo build`, so skip this step.

ihuguet commented on CONTRIBUTING.md:
```diff
@@ -43,6 +43,19 @@ querying network status and applying network states.
+- Install nmstate `cargo install nmstatectl`
+This will download the necessary dependencies and build the nmstate library and command-line tool
+- Build and compile the packages using `cargo build`
+This will compile the source code and generate the executable files
```
Change executable for binary. Some files are libraries, not executables, for example.

ihuguet commented on CONTRIBUTING.md:
```diff
@@ -43,6 +43,19 @@ querying network status and applying network states.
+- Build and compile the packages using `cargo build`
+This will compile the source code and generate the executable files
+- Run unit test to ensure functionality of the  nmstate library and tools `cargo test`
+- View networking state in yaml format `nmstatectl show`
+- Check version `nmstatectl --version`
```
To do this, you need to install it first. You have not installed anything from the repository in any step.

beekaz committed:
CONTRIBUTING.md: Add compilation and instalations steps

Signed-off-by: beekaz <sooperfect00@gmail.com>

ihuguet commented on CONTRIBUTING.md:
```diff
@@ -45,16 +45,14 @@ querying network status and applying network states.
+- Build and compile the packages using Cargo. Run the command cargo build in the terminal.  `cargo build`  
+  This will download the necessary dependencies and build the nmstate library and command-line tool and will compile the source code then generate binary.
+- The compiled files will be placed in the target/debug directory
+- To view the networking state in YAML format, you can use  `target/debug/nmstate show`
```
`target/debug/nmstatectl show`, the binary name is wrong

ihuguet commented on CONTRIBUTING.md:
```diff
@@ -45,16 +45,14 @@ querying network status and applying network states.
+- [Fork](https://github.com/nmstate/nmstate.git) and git clone to local machine
+- Change your current directory to the nmstate directory then into rust   `cd nmstate/rust`
+- Build and compile the packages using Cargo. Run the command cargo build in the terminal.  `cargo build`  
+  This will download the necessary dependencies and build the nmstate library and command-line tool and will compile the source code then generate binary.
```
Here you say 3 times almost the same thing :)
Leave it only as "This will download the necessary dependencies and build the nmstate library and command-line tool."

ihuguet commented on CONTRIBUTING.md:
```diff
@@ -45,16 +45,14 @@ querying network status and applying network states.
+- [Install Rust and Cargo for the development of nmstate](https://doc.rust-lang.org/cargo/getting-started/installation.html) or install the development tools from your distribution's repositories
+- [Fork](https://github.com/nmstate/nmstate.git) and git clone to local machine
+- Change your current directory to the nmstate directory then into rust   `cd nmstate/rust`
+- Build and compile the packages using Cargo. Run the command cargo build in the terminal.  `cargo build`  
```
As you provide the exact command, you can omit "Run the command cargo build in the terminal.", it is redundant.

ihuguet commented on CONTRIBUTING.md:
```diff
@@ -45,16 +45,14 @@ querying network status and applying network states.
-- Check version `nmstatectl --version`
+- [Install Rust and Cargo for the development of nmstate](https://doc.rust-lang.org/cargo/getting-started/installation.html) or install the development tools from your distribution's repositories
+- [Fork](https://github.com/nmstate/nmstate.git) and git clone to local machine
+- Change your current directory to the nmstate directory then into rust   `cd nmstate/rust`
```
I'd prefer this sentence a bit shorter, just "Change your current directory: `cd nmstate/rust`" is enough

ihuguet commented on CONTRIBUTING.md:
```diff
@@ -45,16 +45,14 @@ querying network status and applying network states.
 
 ## Installation and Compilation:
```
The order is reversed, first you have to compile and then you can optionally install. I would change the title to "Compile and install from source".

ihuguet reviewed (changes requested):
Good work, the explanation is complete but short.
However, you need to add the explanations for the optional step of installing to the system.

beekaz committed:
CONTRIBUTING.md: Add compilation and instalations steps

Signed-off-by: beekaz <sooperfect00@gmail.com>

ihuguet commented:
When I said that the text was short, I was saying it as a good thing! Better to be concise! Most of the new text that you have added is boilerplate that is not useful at all. Please remove it, and leave the main title as "Compile and install from source".

Regarding the explanations about how to install to the system: what I mean here is explanations about how to install the compiled binaries of nmstate to the system, not installing cargo from the distribution's repositories. Please watch last week presentation to see how to do it, and remember to mention that this is an optional step.

beekaz commented:
> how to install the compiled binaries of nmstate to the system,

Alright, sorry I misunderstood it, I have resolved it

beekaz committed:
CONTRIBUTING.md: Add compilation and instalations steps

Signed-off-by: beekaz <sooperfect00@gmail.com>

beekaz committed:
Delete CHANGELOG

Signed-off-by: beekaz <sooperfect00@gmail.com>

beekaz committed:
Delete VERSION

Signed-off-by: beekaz <sooperfect00@gmail.com>

beekaz committed:
Delete rust/src/cli/service.rs

Signed-off-by: beekaz <sooperfect00@gmail.com>

beekaz committed:
Delete rust/src/python/setup.py

Signed-off-by: beekaz <sooperfect00@gmail.com>

beekaz committed:
Delete rust/src/python/libnmstate/__init__.py

Signed-off-by: beekaz <sooperfect00@gmail.com>

beekaz committed:
Delete rust/src/cli/Cargo.toml

Signed-off-by: beekaz <sooperfect00@gmail.com>

beekaz committed:
Delete rust/src/clib/Cargo.toml

Signed-off-by: beekaz <sooperfect00@gmail.com>

beekaz committed:
Delete rust/src/lib/Cargo.toml

Signed-off-by: beekaz <sooperfect00@gmail.com>

ffmancera commented:
I think this PR is completely out of the scope. The commits are removing essential files for the project.
